### DISCUSSION TITLE
Enable Broadcast

### DISCUSSION THREAD

**Discussion by TobyTredger** [Networking-for-Arduino/EthernetENC#68] [Q&A]
Thank you so much for this lib. It's the only one I can get working with a pi pico. 

However I really need broadcast for my project and I know it's disabled.

What would be required to enable reception of broadcast packets?

I've tried enabling broadcast packages in filtering using this

`  writeReg(ERXFCON, ERXFCON_BCEN|ERXFCON_UCEN|ERXFCON_CRCEN|ERXFCON_PMEN);
`

And i've checked the 

`#define UIP_CONF_BROADCAST    1`

I'm still not getting anything when sending broadcast data on the network.

I've tried binding to xx.xx.xx.255 and 0.0.0.0.

Any ideas would be very much appreciated. :)

**Comment by JAndrassy** [comment]
with PI Pico Arduino use the library bundled with the RP2040 Arduino core

**Reply by TobyTredger** [reply]
Thanks for the response. That would require me to refactor all my codebase. Is it really that hard to enable broadcast ? Are there other advantages to using the RP2040 arduino core ethernet support ?

**Reply by JAndrassy** [reply]
the EthernetCompat.h has the same API as this library. https://github.com/earlephilhower/arduino-pico/blob/master/libraries/lwIP_Ethernet/examples/EthernetLibCompatible/EthernetLibCompatible.ino
the performance is better

**Reply by TobyTredger** [reply]
For the EthernetClient and server, I agree. But i'm struggling to setup the boad. I can no longer do this...

    Ethernet.init(17);

    Ethernet.begin(artnetNodeConfig.mac, {192,168,1,255}, gateway, gateway, artnetNodeConfig.mask);

**Reply by TobyTredger** [reply]
In fact having followed their documentation, I cant get the interface working with the ENC28J60

**Reply by JAndrassy** [reply]
don't use address .255

**Reply by TobyTredger** [reply]
Ok I simply cannot get the built in ethernet lib working with the ENC28J60. It it really that hard to handle broadcast packets ?

**Reply by JAndrassy** [reply]
don't assign IP address with  .255

**Reply by TobyTredger** [reply]
That doesn't help.

**Reply by TobyTredger** [reply]
I can get an IP address, but I don't get anything through using the TCP Server. Even just using the basic example code.

**Reply by JAndrassy** [reply]
it doesn't help for EthernetENC broadcast or for lwip_ens28j60?

**Reply by TobyTredger** [reply]
I followed this doc to get ethernet setup and a tcp server running. Although I can get an ip, I can't get anything through on the server.
https://arduino-pico.readthedocs.io/en/latest/ethernet.html

**Reply by JAndrassy** [reply]
maybe because your sketch expects proper server.available() and that library only has server.accept() (and available() works as accept()).

**Comment by TobyTredger** [comment]
Thanks for the reply. I tried that. Using server.accept didn’t help.

**Reply by JAndrassy** [reply]
how could it. that is the problem. but you can work around it modifying the sketch. don't destroy the client returned by accept(), because that closes the connection

**Reply by TobyTredger** [reply]
This gives me nothing
```

    EthernetClient client = tcp.accept();

    if (client) {
        Serial.println("CONNECTION.");
```

**Reply by TobyTredger** [reply]
I setup using this globaly.

`ArduinoENC28J60lwIP Ethernet = ArduinoENC28J60lwIP();
` 

Then:

`    Ethernet.begin(artnetNodeConfig.mac, {192,168,1,71}, gateway, gateway, artnetNodeConfig.mask);`

**Reply by JAndrassy** [reply]
`EthernetClient client = tcp.accept();` is you problem. make client global and only call accept if the client is disconnected
